### ISSUE TITLE
elasticsearch pip package not found during upgrade

### ISSUE THREAD

**Issue by BjoernT** [rcbops/rpc-openstack#1092]
##### What version (commit) do you encounter this issue on?

r12.1.1
##### What type of issue is this (bug, feature, documentation, regression)?

Bug
##### What is the expected behavior?

Install desired version
##### What is actually happening?

Per #902 we limit the version of elasticsearch <2.1.0 but in the repo is only 2.1.0 available (http://rpc-repo.rackspace.com/os-releases/liberty/) hence we should do a <= instead?
Additionally I do not see even elasticsearch-curator as pip package

Seems like the desired versions are elasticsearch-2.0.0 and elasticsearch_curator-3.4.1 but I could not find them in my repo container, so we have to look where they are even coming from
##### How can we reproduce this issue?

pip install
##### How can we verify your issue is addressed?

Mismatch resolved
##### If this is a large issue, what can be done immediately to alleviate the issue while working on a longer-term solution?

Normal
##### What is your evaluation on the importance of this issue, and why?

N/A
###### Remove branches (lines) you do not wish to target.
- [ ] Master
- [ ] liberty-12.2
- [ ] liberty-12.1

**Comment by git-harry** [comment]
@BjoernT I don't believe there is an issue here to address. Liberty does a repo-build not a repo-clone. In addition the clone in Kilo is done based on a tag not the head of the branch. We have periodic and commit jobs that test installing elasticsearch and they currently pass, e.g. [1].

If you are hitting an issue please provide a more complete account of what you see and the steps required to reproduce the issue. The issue template is there to help guide you towards providing us with the information we need to assist you and not simply a box-ticking exercise.

Based on what I've said I will close this issue but please reopen it if you are able to provide further details.

[1] http://jenkins.propter.net/job/RPC-AIO/6181/consoleFull

**Comment by BjoernT** [comment]
@git-harry  Obviously I have a issue, otherwise I would not open a bug. My issues is that there is only elasticsearch 2.1.0 inside the os-repo on the repo container so the pip installs on elasticsearch and curator are failing. So please reopen.

**Comment by git-harry** [comment]
@BjoernT you are not giving details of your issue you are only saying what you believe the issue is and how to resolve it. As I explained the elasticsearch wheel should not need to be on rpc-repo for the repo to build. If you can supply exact steps to reproduce the issue, logs showing where Ansible fails and any other relevant data this issue can be reviewed. As it stands r12.1.1 passes our testing [1] and without further information I cannot justify further investigation.

I am happy to reopen the issue but will mark it as requiring further information until you have supplied the necessary details.

[1] http://jenkins.propter.net/job/RPC-AIO/5553/consoleFull

**Comment by BjoernT** [comment]
@git-harry Yeah and you close bugs without giving someone the chance to add information, how about just asking first what issue I might have. I have just reported what I think the issue is, since I do not see versions > 2.1.0 in local build repos either, no matter if we sync from liberty or not.

**Comment by BjoernT** [comment]
Seems like the desired versions are elasticsearch-2.0.0 and elasticsearch_curator-3.4.1 but I could not find them in my repo container, so I have to look where they are even coming from

**Comment by d34dh0r53** [comment]
I think this may be an issue, the version requirement does not seem correct to me.

**Comment by alextricity25** [comment]
@BjoernT, our liberty AIO built a repo server locally using the repo-build process. Currently we see elasticsearch-2.0.0-py2.py3-none-any.whl and elasticsearch_curator-3.4.1-py2.py3-none-any.whl. Which is correct and doesn't present us with issues.
The wheels in rpc-repo are not correct like you say, but it's my understanding that we no longer clone from this mirror, so it is irrelevant.

You mentioned upgrades in the bug title. What kind of upgrade were you doing? From what release? We would like to test the upgrade process to make sure the repo-server is re-building correctly.

**Comment by cloader89** [comment]
@BjoernT  which version of Kilo should we upgrade from?

**Comment by cloader89** [comment]
I  spoke with @Bjoern yesterday and he was going to try to see what version of Kilo we needed to use. He also mentioned it might no longer be an issue. We might close it -- and re-open if it is still an issue

**Comment by b3rnard0** [comment]
@claco @git-harry Potentially not an issue any more -- thoughts?

**Comment by BjoernT** [comment]
Closing unless we encounter it

**Comment by BjoernT** [comment]
This just happened during an upgrade again, although es was capped to <2.1.0 during repo build process:

```
        "role_packages": {
            "beaver": [
                "beaver==33.3.0"
            ],
            "elasticsearch": [
                "elasticsearch-curator<3.5.0",
                "elasticsearch<2.1.0"
            ],
```

interestingly I saw those messages during repo build :

````skipping: [infra02_repo_container-66e78eb4] => (item={'key': 'elasticsearch', 'value': ['elasticsearch-curator<3.5.0', 'elasticsearch<2.1.0']})```
